receiver/otlpreceiver/otlp.go

```diff
@@ -180,15 +205,21 @@ func (r *otlpReceiver) Shutdown(ctx context.Context) error {
 		if r.serverGRPC != nil {
 			r.serverGRPC.GracefulStop()
 		}
+
+		r.shutdownWG.Wait()
```

Comment:
This is not going to work well. If the same `otlpReceiver` is reused then the code from line 198 will be executed only first time you call shutdown.

I think instead of trying to reuse the same instance of `otlpReceiver` we should remove that from the factory global map when the instance is shutdown (a.k.a in this stopOnce loop).

Reply:
Same happens for start as well.

Reply:
The question about the same OTLP receiver on different data type pipelines is an interesting one: Start/Shutdown won't fail but when the first call is made it starts/stops receiving for all pipelines since the servers are shared... this is not an issue because the start is done by component type not by pipeline. Anyway, the receiver should be removed from the map on Shutdown. Done.